srcs/runners_manager/vm_creation/openstack.py

```diff
@@ -68,38 +68,63 @@ def create_vm(self, runner: Runner, runner_token: int or None,
                   github_organization: str, installer: str, call_number=0):
         """
         TODO `tenantnetwork1` is a hardcoded network we should put this in config later on
+        Every call with nova_client looks very unstable.
         """
 
         if call_number > 5:
             return None
 
-        sec_group_id = self.neutron.list_security_groups()['security_groups'][0]['id']
-        nic = {'net-id': self.neutron.list_networks(name='tenantnetwork1')['networks'][0]['id']}
-        instance = self.nova_client.servers.create(
-            name=runner.name, image=self.nova_client.glance.find_image(runner.vm_type.image),
-            flavor=self.nova_client.flavors.find(name=runner.vm_type.flavor),
-            security_groups=[sec_group_id], nics=[nic],
-            userdata=self.script_init_runner(runner, runner_token, github_organization,
-                                             installer)
-        )
-        while instance.status not in ['ACTIVE', 'ERROR']:
-            instance = self.nova_client.servers.get(instance.id)
-            time.sleep(2)
-        if instance.status == 'ERROR':
-            logger.info('vm failed, creating a new one')
-            self.delete_vm(instance.id)
-            time.sleep(2)
+        # Delete all VMs with the same name
+        vm_list = self.nova_client.servers.list(search_opts={'name': runner.name},
+                                                sort_keys=['created_at'])
+        for vm_id in vm_list:
+            self.nova_client.servers.delete(vm_id)
+
+        instance = None
+        try:
+
+            sec_group_id = self.neutron.list_security_groups()['security_groups'][0]['id']
+            nic = {'net-id': self.neutron.list_networks(name='tenantnetwork1')['networks'][0]['id']}
+            image = self.nova_client.glance.find_image(runner.vm_type.image)
+            flavor = self.nova_client.flavors.find(name=runner.vm_type.flavor)
+
+            instance = self.nova_client.servers.create(
+                name=runner.name,
+                image=image,
+                flavor=flavor,
+                security_groups=[sec_group_id], nics=[nic],
+                userdata=self.script_init_runner(runner, runner_token, github_organization,
+                                                 installer)
+            )
+
+            while instance.status not in ['ACTIVE', 'ERROR']:
+                instance = self.nova_client.servers.get(instance.id)
+                time.sleep(2)
+
+            if instance.status == 'ERROR':
+                logger.info('vm failed, creating a new one')
```

Comment:
Bonus point if you can add the instance.id here, that would be useful to share with support to track what happens to the creation of those vms id.